nautobot/core/factory.py

```diff
@@ -2,13 +2,28 @@
 import factory
 from factory.django import DjangoModelFactory
 import factory.random
+import itertools
 
 from nautobot.extras.models import Tag
 
 
+class UniqueFaker(factory.Faker):
+    """https://github.com/FactoryBoy/factory_boy/pull/820#issuecomment-1004802669"""
+
+    @classmethod
+    def _get_faker(cls, locale=None):
+        return super()._get_faker(locale=locale).unique
+
+    def clear(self, locale=None):
+        subfaker = self._get_faker(locale)
+        subfaker.clear()
+
+
 class BaseModelFactory(DjangoModelFactory):
     """Base class for all Nautobot model factories."""
 
+    id = UniqueFaker("uuid4")
```

Comment:
Huh. Making the factory UUIDs deterministic? What was the driver for this? I worry that it may mask some bad test practices if we rely on specific UUIDs being present.

Reply:
The only reason I added this was to make sure that my factory changes didn't break the determinism of the factories (which I did on my first couple passes). It also makes the diffs easier to manage if we ever decide to generate fixtures and use those instead of running the factories every time (as in #3255). I'm fine with removing this but I don't know if I agree with leaving it out to enforce better coding practices.